synapse/lib/stormlib/gen.py

```diff
@@ -66,23 +97,221 @@ class LibGen(s_stormtypes.Lib):
             [ ou:industry=* :name=$name ]
             return($node)
         }
+
         function newsByUrl(url) {
             media:news:url=$url
             return($node)
             [ media:news=* :url=$url ]
             return($node)
         }
+
         function softByName(name) {
             it:prod:softname=$name
             -> it:prod:soft
             return($node)
             [ it:prod:soft=* :name=$name ]
             return($node)
         }
+
         function vulnByCve(cve) {
             risk:vuln:cve=$cve
             return($node)
             [ risk:vuln=* :cve=$cve ]
             return($node)
         }
+
+        function riskThreat(name, reporter) {
+            ou:name=$name
+            tee { -> risk:threat:org:name } { -> risk:threat:org:names } |
+            +:reporter:name=$reporter
+            return($node)
+
+            [ risk:threat=*
+                :org:name=$name
+                :reporter = { yield $orgByName($name) }
+                :reporter:name = $reporter
+            ]
+            return($node)
+        }
+
+        function riskToolSoftware(name, reporter) {
+
+            it:prod:softname = $name
+            -> risk:tool:software
+            +:reporter:name = $reporter
+            { -:reporter [ :reporter=$orgByName($reporter) ] }
+            return($node)
+
+            [ risk:tool:software=*
+                :soft:name = $name
+                :reporter:name = $reporter
+                :reporter = { yield $orgByName($reporter) }
+            ]
+
+            return($node)
+        }
+
+        function psContactByEmail(type, email) {
+            ps:contact:email = $email
+            +:type = $type
+            return($node)
+            [ ps:contact=*
+                :email = $email
+                :type = $type
+            ]
+            return($node)
+        }
+
+        function psContactByName(type, name) {
+            ps:contact:email = $email
+            +:type = $type
+            return($node)
+            [ ps:contact=*
+                :email = $email
+                :type = $type
+            ]
+            return($node)
+        }
+
+        function polCountryByIso2(iso2) {
+            pol:country:iso2=$iso2
+            return($node)
+            [ pol:country=* :iso2=$iso2 ]
+            return($node)
+        }
     '''
+
+stormcmds = (
+
+    {
+        'name': 'gen.ou.org',
+        'descr': 'Lift (or create) an ou:org node based on the organization name.',
+        'cmdargs': (
+            ('name', {'help': 'The name of the organization.'}),
+        ),
+        'storm': 'yield $lib.gen.orgByName($cmdopts.name)',
+    },
+    {
+        'name': 'gen.ou.org.hq',
```

Comment:
Since this is creating a ps:contact node, do we want to consider renaming to "gen.org.ps.contact" or "gen.ps.contact.org"?

Reply:
The naming convention i'm going with for these is based on the secondary property name to make it clear that it is explicitly *that*. Will be useful in future cases where there could be more than one property of a given form.